src/System.Private.CoreLib/src/System/Resources/ManifestBasedResourceGroveler.cs

```diff
@@ -486,11 +474,45 @@ private void HandleResourceStreamMissing(string fileName)
             if (_mediator.LocationInfo != null && _mediator.LocationInfo.Namespace != null)
                 resName = _mediator.LocationInfo.Namespace + Type.Delimiter;
             resName += fileName;
-            throw new MissingManifestResourceException(SR.Format(SR.MissingManifestResource_NoNeutralAsm, resName, _mediator.MainAssembly.GetSimpleName()));
+            throw new MissingManifestResourceException(SR.Format(SR.MissingManifestResource_NoNeutralAsm, resName, _mediator.MainAssemblySimpleName));
+        }
+
+        // Internal version of GetSatelliteAssembly that avoids throwing FileLoadException
+        private Assembly InternalGetSatelliteAssembly(Assembly mainAssembly,
+                                                      CultureInfo culture,
+                                                      Version version)
+        {
+            if (mainAssembly is RuntimeAssembly runtimeMainAssembly)
+            {
+                return runtimeMainAssembly.InternalGetSatelliteAssembly(culture, version, throwOnFileNotFound: false);
+            }
+
+            try
+            {
+                return mainAssembly.GetSatelliteAssembly(culture, version);
+            }
+            catch (FileLoadException)
+            {
+                return null;
+            }
         }
 
         [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
         [return: MarshalAs(UnmanagedType.Bool)]
         internal static extern bool GetNeutralResourcesLanguageAttribute(RuntimeAssembly assemblyHandle, StringHandleOnStack cultureName, out short fallbackLocation);
+
+        internal static bool GetNeutralResourcesLanguageAttribute(Assembly assemblyHandle, ref string cultureName, out short fallbackLocation)
+        {
+            if (assemblyHandle is RuntimeAssembly runtimeAssemblyHandle)
+            {
+                return GetNeutralResourcesLanguageAttribute(runtimeAssemblyHandle.GetNativeHandle(),
+                                                            JitHelpers.GetStringHandleOnStack(ref cultureName),
+                                                            out fallbackLocation);
+            }
+
+            // Not currently supported on custom Assembly types or AssemblyBuilder            
+            fallbackLocation = 0;
+            return false;
```

Comment:
This case could be implemented in the same way as in CoreRT. Not sure if it's worth it though since it was not supported before.